File: uLab/Assets/Scripts/Test/BarWindow.cs


using Locke;
using Locke.ui;
using UnityEngine.UI;

public class BarWindow : IWindow
{
	protected override void OnEnter(IContext context)
	{
		Button button = this.FindWidget<Button>("buttons/button1");
		button.transform.GetComponentInChildren<Text>().text = "I'm bar.";
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

}

File: uLab/Assets/Lua/UI/BarWindow.lua

local gameObject
local transform

BarWindow = class("BagWindow",UIBase)
--local this = BarWindow

function BarWindow.OnInit(obj)
	gameObject = obj
	transform = obj.transform
	--LogInfo("Bar OnInit")
end

function BarWindow:OnEnter()
	--LogInfo("Bar OnEnter")
end

function BarWindow:OnExit()
	--LogInfo("Bar OnExit")
end

function BarWindow:OnPause()
	--LogInfo("Bar OnPause")
end

function BarWindow:OnResume()
	--LogInfo("Bar OnResume")
end
